website/docs/developer/research/research-reconciliation.mdx

```diff
@@ -0,0 +1,683 @@
+
+- [GraphQL Servers (prebuilt)](https://www.moesif.com/blog/graphql/technical/Ways-To-Add-GraphQL-To-Your-Postgres-Database-Comparing-Hasura-Prisma-and-Others/)
+- [Build your Own GraphQL Server](https://www.howtographql.com/choose/)
+- [GraphQL classes](https://www.edx.org/course/exploring-graphql-a-query-language-for-apis)
+- [GraphQL Example with CMR](https://git.earthdata.nasa.gov/projects/CMRQL/repos/hackfest-cmr-graphql/browse)
+- [Using Apollo Server as a Lambda](https://www.apollographql.com/docs/apollo-server/deployment/lambda/)
+
+
+### SQS
+
+[AWS SQS FIFO queues](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/FIFO-queues.html)
+should be used to manage the records to be written to the ORCA database. This
+allows the team to decouple the database code from several lambdas and centralize
+the code to one area. This may or may not be needed if GraphQL is utilized.
+
+
+### AWS PostgreSQL RDS Offerings
+
+To better align with Cumulus, ORCA will utilize the AWS PostgreSQL RDS offerings.
+See notes on the ORCA transition to the ORCA database [here](https://wiki.earthdata.nasa.gov/display/ORCA/ORCA+Migration+to+Cumulus+RDS).
+The resultant code from the reconciliation work should be able to handle working
+with a PostgreSQL backed. Both PostgreSQL RDS and Amazon Aurora are viable back end
+choices that may be utilized. See a comparison article on the databases
+[here](https://aws.amazon.com/blogs/database/is-amazon-rds-for-postgresql-or-amazon-aurora-postgresql-a-better-choice-for-me/).
+
+
+### NodeJS (Knex) / Python (SQLAlchemy)
+
+An agnostic language framework for working with the database should be used some
+options include but are not limited to the items described below. Utilizing a
+more general API may help manage changes to the back end database engine with
+little or no code changes.
+
+- [Nodejs and Knex](http://knexjs.org/) is what the Cumulus team currently
+  utilizes for database interactions. Implementation could follow best practices
+  already established by the Cumulus team.
+
+- [Python and SQLAlchemy Core](https://docs.sqlalchemy.org/en/14/core/) is currently
+  being used in some ORCA code.
+
+
+## Comparison between ORCA S3 and the ORCA Catalog
+
+In addition to the comparison between the Cumulus catalog and ORCA, ORCA will
+also need to manage and maintain the catalog holdings between the ORCA database
+and the actual contents of the ORCA S3 buckets. This will require comparing the
+*Key*, *LastModified*, *ETag*, and *Size* keys to the of the files in the bucket
+to the ORCA holdings catalog. Discrepancies should be reported back to the user
+and detail orphan and missing files. We could possibly create a temporary reporting
+table to hold these values that the Cumulus Dashboard could display for us.
+
+Some things to think through when designing this solution are factors of scale
+and managing them. To provide some thought and guidance, LP DAAC one of the larger
```

Comment:
<!--StartFragment-->and managing them. To provide some thought and guidance, LP DAAC (one of the largerDAACs) has approximately 300 million files in the archive and growing. If all the<!--EndFragment-->

Reply:
fixed.